### PR TITLE
Edit delete

### PR DESCRIPTION
**Pull request by jwilcox76**
pull request for the create/delete branch

### CODE REVIEW

File: routes/tweet_routes.py [hunk 1]
```diff
@@ -119,3 +121,49 @@ def create_comment(tweet_id):
...
+
+@bp.delete("/delete/<int:tweet_id>")
+def delete_message(tweet_id):
+    userID = request.cookies.get('userId')
+    if not userID:
+        return "<script>window.location = '/'</script>"
+
+    tweet = Tweet.find(tweet_id)
+
+    if tweet.user.userID != userID:
```
Comment (BurginLuker) [inline routes/tweet_routes.py]: Might be wise to check the tweet user compared to the user using User.find(userID). This is a deeper equality and safer, but If that does not work then I guess this is fine

File: routes/tweet_routes.py [hunk 2]
```diff
@@ -119,3 +121,49 @@ def create_comment(tweet_id):
...
+
+
+@bp.get("/tweets/editMessage/<editing>/<int:tweet_id>")
+def edit_message(editing, tweet_id):
+    userID = request.cookies.get('userId')
+    if not userID:
+        return "<script>window.location = '/'</script>"
+    tweet = Tweet.find(tweet_id)
+
+    if tweet.user.userID != userID:
```
Comment (BurginLuker) [inline routes/tweet_routes.py]: See above

File: templates/tweet.html [hunk 3]
```diff
@@ -9,12 +10,46 @@ <h3 style="font-size: 18px; color: #f0f1f3; margin: 0; font-weight: bold;">
...
+                        <button hx-get="/tweets/editMessage/{{True}}/{{tweet.tweet.id}}"
+                                hx-target="#tweet-text-{{ tweet.tweet.id }}"
+                                hx-swap="innerHTML">
+                            EDIT
+                        </button>
+
+                        <form hx-delete="/delete/{{tweet.tweet.id}}"
+                              hx-target="#tweetContainer-{{tweet.tweet.id}}"
+                              hx-swap="outerHTML">
+                            <!--                            <input type="hidden" name="message" value="{{tweet.tweet.message}}">-->
```
Comment (BurginLuker) [inline templates/tweet.html]: Remove

File: src/model/tweet.py [hunk 4]
```diff
@@ -9,7 +9,8 @@
 class Tweet(BaseModel):
     message = FixedCharField(max_length=300)
     user = ForeignKeyField(User, backref='tweets')
-    post_date = DateTimeField(default=lambda: datetime.datetime.now(ZoneInfo("America/Denver")).strftime('%b %d, %Y - %I:%M %p'))
+    # post_date = DateTimeField(default=lambda: datetime.datetime.now(ZoneInfo("America/Denver")).strftime('%b %d, %Y - %I:%M %p'))
```
Comment (BurginLuker) [inline src/model/tweet.py]: remove

File: routes/__pycache__/user_routes.cpython-313.pyc
Comment (BurginLuker) [inline routes/__pycache__/user_routes.cpython-313.pyc:1]: Not sure why we are committing cache?

Comment (jwilcox76) [reply routes/__pycache__/user_routes.cpython-313.pyc:1]: Not sure why that happened, my b

**Comment by Tcozzie** [comment]
PNG image

**Comment by BurginLuker** [comment]
Screenshot 2024-12-02 at 8 11 48 PM
Why does the delete button have so much extra space below it, also when I edit a tweet why do 5 empty spaces get added before the original text

**Comment by Tcozzie** [comment]
can you please uncommon the cache files

**Comment by BurginLuker** [comment]
Screenshot 2024-12-02 at 8 17 04 PM
No character counter leads to strange behavior!

**Comment by Tcozzie** [comment]
you have to delete the likes in the redis cache as well.
same issue with the comments

**Comment by BurginLuker** [comment]
I dont think we should be able to edit or delete from the homepage, you should only be able to edit from the create/edit page

**Comment by Tcozzie** [comment]
and when the edit button is clicked can the dropdown close. also if the user clicks outside of the dropdown can it close

**Comment by BurginLuker** [comment]
> and when the edit button is clicked can the dropdown close. also if the user clicks outside of the dropdown can it close

Yes, you should not be able to have multiple open at once

**Comment by Tcozzie** [comment]
> I dont think we should be able to edit or delete from the homepage, you should only be able to edit from the create/edit page

I also dont think you should be able to edit multiple tweets a once

File: templates/tweet.html [hunk 5]
```diff
@@ -9,12 +10,45 @@ <h3 style="font-size: 18px; color: #f0f1f3; margin: 0; font-weight: bold;">
...
             <span class="time" style="color: #4f8a8b; font-size: 15px;">{{tweet.tweet.post_date}}</span>
+
+            {% if tweet.tweet.user.userID == user.userID %}
+            <div style=" position: relative;font-size: 12px; cursor: pointer; text-align: right; margin-left: 5px; display: flex; flex-direction: column; align-items: center;">
+                <div class="dropdown">
+                    <button onclick="toggleDropdown(event,this)"
+                            style="margin-top: 2px; background: none; border: none; font-size: 10px; display: inline-block; outline: none;">
+                        <span style="display: block; height: 4px; width: 8px; margin: 2px 0; background-color: #4f8a8b; border-radius: 2px;"></span>
+                        <span style="display: block; height: 4px; width: 8px; margin: 4px 0; background-color: #4f8a8b; border-radius: 2px;"></span>
+                        <span style="display: block; height: 4px; width: 8px; margin: 4px 0; background-color: #4f8a8b; border-radius: 2px;"></span>
```
Comment (Tcozzie) [inline templates/tweet.html]: could you remove these spans and use the three dot svg that we have in images?

**Comment by jwilcox76** [comment]
Newest push should have all the fixes for all the suggestions you guys wanted

File: templates/editMessage.html [hunk 6]
```diff
@@ -0,0 +1,11 @@
+    <form id="formID-{{tweet.id}}"
+          hx-post="/tweets/{{ tweet.id }}/update"
+          hx-swap="outerHTML"
+          hx-target="#tweet-text-{{ tweet.id }}"
+          style="font-size: 18px; line-height: 1.6; font-weight: bold; color: #f0f1f3;">
+        <textarea autofocus required rows="1" maxlength="300" type="text" name="message" style="border: white">{{tweet.message.strip()}}
+        </textarea>
+        <button type="submit" style="background-color: #4f8a8b; border: white; color: white">
+            update
```
Comment (BurginLuker) [inline templates/editMessage.html]: `Update`

Comment (BurginLuker) [inline templates/editMessage.html]:
```suggestion
            Update
```

**Pull request merged**